awss3/store.go

```diff
@@ -107,7 +109,24 @@ type (
 // NewClient create new AWS s3 Client.  Uses cloudstorage.Config to read
 // necessary config settings such as bucket, region, auth.
 func NewClient(conf *cloudstorage.Config) (*s3.S3, *session.Session, error) {
+	awsConf, err := makeAWSConf(conf)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	sess := session.New(awsConf)
+	if sess == nil {
+		return nil, nil, ErrNoS3Session
+	}
+
+	s3Client := s3.New(sess)
+
+	ensureBucket(s3Client, conf.Bucket)
```

Comment:
@sergeyt shouldn't you be checking the error returned from this function?

Reply:
@epsniff yes, but only network connections I guess. if bucket exists should be no error